src/qpython.cpp

```diff
@@ -127,6 +129,103 @@ QPython::addImportPath(QString path)
+{
+    // The plan is:
+    // import the module, given by module_name,
+    // don't put it into globals of priv,
+    // make some temporary dict with globals + the new module,
+    // get the callables from the temp globals, given by names in args,
+    // put the callables into globals of priv
+    // call it a day
+    // (then one could reimplement importModule based on importNames)
+
+    // getting the module (the code is from importModule_sync)
+    QByteArray utf8bytes = module_name.toUtf8();
+    const char *moduleName = utf8bytes.constData();
+
+    ENSURE_GIL_STATE;
+
+    bool use_api_10 = (api_version_major == 1 && api_version_minor == 0);
+
+    PyObjectRef module;
+
+    if (use_api_10) {
+        // PyOtherSide API 1.0 behavior (star import)
+        module = PyObjectRef(PyImport_ImportModule(moduleName), true);
+    } else {
+        // PyOtherSide API 1.2 behavior: "import x.y.z"
+        PyObjectRef fromList(PyList_New(0), true);
+        module = PyObjectRef(PyImport_ImportModuleEx(const_cast<char *>(moduleName),
+                    NULL, NULL, fromList.borrow()), true);
+    }
+
+    if (!module) {
+        emitError(QString("Cannot import module: %1 (%2)").arg(module_name).arg(priv->formatExc()));
+        return false;
+    }
+
+    if (!use_api_10) {
+        // PyOtherSide API 1.2 behavior: "import x.y.z"
+        // If "x.y.z" is imported, we need to set "x" in globals
+        if (module_name.indexOf('.') != -1) {
+            module_name = module_name.mid(0, module_name.indexOf('.'));
+            utf8bytes = module_name.toUtf8();
+            moduleName = utf8bytes.constData();
+        }
+    }
+
+    // so at this point we got the module in PyObjectRef module,
+    // it should be imported in Python (on the heap etc)
+    // and we've got its' name in const char *moduleName
+
+    // make the temporary globals:
+    PyObjectRef globals_temp = PyObjectRef(PyDict_New(), true);
+    int err = PyDict_Update(globals_temp.borrow(), priv->globals.borrow());
+    if (err != 0) {
+        emitError(QString("Error in creating temporary globals namespace during importNames"));
+        return false;
+    }
+    PyDict_SetItemString(globals_temp.borrow(), moduleName, module.borrow());
+
+    // Get the names of functions/objects to import
+    QVariantList vl = args.toList();
+
+    QString obj_name;   // object name to import
+    QString full_name;  // full name of the object == module_name.obj_name
+    PyObjectRef result; // the object, obtained from globals_temp
+
+    for (QVariantList::const_iterator obj = vl.begin(); obj != vl.end(); ++obj)
+    {
+        obj_name = obj->toString();
+        // each obj in the list has to be a function name, not an expression
+        // however, further PyRun_String is used to get the object
+        full_name = module_name + "." + obj_name;
+        utf8bytes = full_name.toUtf8();
+        PyObject *res = PyRun_String(utf8bytes.constData(),
```

Comment:
Or actually what you want is probably `PyObject_GetAttrString` (which takes a `char *`).